src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -112,44 +153,331 @@ class AppsController(
+      provide(t)
+    case Failure(ValidationFailedException(_, failure)) =>
+      reject(EntityMarshallers.ValidationFailed(failure))
+    case Failure(AccessDeniedException(msg)) =>
+      reject(AuthDirectives.NotAuthorized(HttpPluginFacade.response(authorizer.handleNotAuthorized(identity, _))))
+    case Failure(_: AppNotFoundException) =>
+      reject(
+        maybeAppId.map { appId =>
+          Rejections.EntityNotFound.app(appId)
+        } getOrElse Rejections.EntityNotFound()
+      )
+    case Failure(RejectionError(rejection)) =>
+      reject(rejection)
+    case Failure(ex) =>
+      throw ex
+  }
+
+  private def patchSingle(appId: PathId)(implicit identity: Identity) =
+    update(appId, partialUpdate = true, allowCreation = false)
+
+  private def putSingle(appId: PathId)(implicit identity: Identity) =
+    parameter('partialUpdate.as[Boolean].?(true)) { partialUpdate =>
+      update(appId, partialUpdate = partialUpdate, allowCreation = true)
+    }
+
+  private def deleteSingle(appId: PathId)(implicit identity: Identity) =
+    forceParameter { force =>
+      lazy val notFound: Either[Rejection, RootGroup] =
+        Left(Rejections.EntityNotFound.app(appId))
+
+      lazy val notAuthorized: Either[Rejection, RootGroup] =
+        Left(NotAuthorized(HttpPluginFacade.response(authorizer.handleNotAuthorized(identity, _))))
+
+      def deleteApp(rootGroup: RootGroup): Either[Rejection, RootGroup] = {
+        rootGroup.app(appId) match {
+          case None =>
+            notFound
+          case Some(app) =>
+            if (authorizer.isAuthorized(identity, DeleteRunSpec, app))
+              Right(rootGroup.removeApp(appId))
+            else
+              notAuthorized
+        }
+      }
+
+      rejectingLeft(onSuccess(groupManager.updateRootEither(appId.parent, deleteApp, force = force))) { plan =>
+        completeWithDeploymentForApp(appId, plan)
+      }
+    }
+
+  private def rejectLeftViaThrow[T](t: Either[Rejection, T]): T = t match {
+    case Left(r) => throw RejectionError(r)
+    case Right(t) => t
+  }
+
+  private def restartApp(appId: PathId)(implicit identity: Identity) = {
+    forceParameter { force =>
+
+      def markForRestartingOrThrow(opt: Option[AppDefinition]): Either[Rejection, AppDefinition] =
+        opt.map(Right(_)).getOrElse(Left(Rejections.EntityNotFound.app(appId): Rejection))
+          .flatMap { checkAuthorization(UpdateRunSpec, _) }
+          .map(_.markedForRestarting)
+
+      val newVersion = clock.now()
+
+      onSuccessLegacy(Some(appId))(
+        groupManager.updateApp(
+          appId,
+          { app => rejectLeftViaThrow(markForRestartingOrThrow(app)) },
+          newVersion, force)
+      ).apply { restartDeployment =>
+          completeWithDeploymentForApp(appId, restartDeployment)
+        }
+    }
+  }
+
+  private def listRunningTasks(appId: PathId)(implicit identity: Identity) = {
+    val maybeApp = groupManager.app(appId)
+
+    maybeApp map { app =>
```

Comment:
I think we have/had a style guideline to write `object.method {`, not `object map {`, if I recall correctly.